protocol/0014-ORDT-order_types.md

```diff
@@ -90,6 +90,76 @@ Network orders are used during [position resolution](./0012-POSR-position_resolu
+
+### Iceberg Order Batch Submission
+
+1. For multiple iceberg orders submitted as a batch of orders with a mix of ordinary limit orders and market orders, the iceberg orders are processed atomically and the order book volume and price, margin calculations , order status are all correct - What else needs checking ?
+2. What other scenarios need testing ?
+
+### Iceberg Order Submission - Negative tests
+
+1. An iceberg order with a non persistent TIF (IOC, FOK) is rejected with a valid error message.
+2. An iceberg market order with any TIF is rejected with a valid error message.
+3. A reduce-only iceberg order with any TIF is rejected with a valid error message.
+4. An iceberg order with initial peak size less than the minimum order size rejected with a valid error message.
+5. An iceberg order with initial peak size greater than the total order size is rejected with a valid error message.
+6. An iceberg order with minimum peak size less than 0 is rejected with a valid error message.
+7. An iceberg order with minimum peak size greater than initial peak size is rejected with a valid error message.
+
+### Iceberg Order Amendment
+
+1. Amending an iceberg order to increase size will increase the total and remaining quantities of the order and time priority of the order is not lost.
+2. Amending an iceberg order to decrease size will decrease the total and remaining quantities and time priority of the order is not lost.
+3. Amend an iceberg order to decrease size so that the displayed quantity is decreased. Total and remaining quantity is decreased, margin is recalculated and released and time priority is not lost.
+4. What other scenarios do we need to consider here ?
+
+### Iceberg Order Cancellation
+
+1. Cancelling an iceberg order will cancel the order, remove it from the order book , release margin and update order book to reflect the change.
+
+### Iceberg Order Execution
+
+1. An aggressive iceberg order that crosses with an order where volume > iceberg volume, the iceberg order gets fully filled on entry, the iceberg order status is filled, the remaining quantity = 0. Are atomic trades generated OR one single trade for each display quantity volume ???
+2. An aggressive iceberg order that crosses with an order where volume < iceberg volume. The initial display quantity is filled and the remaining volume is unfilled. Status of iceberg order is partially filled , the volume remaining = (quantity - initial volume) and the remaining volume sits on the book. When additional orders thar are submitted which consume the remaining volume on the iceberg order , the volume of the iceberg order is refreshed as and when the volume dips below the minimum peak size.
+3. A passive iceberg order (the only order at a particular price level) when crossed with another order that comes in which consumes the full volume of the iceberg order is fully filled. Status of iceberg order is filled and the remaining = 0. Atomic trades are produced.
```

Comment:
@davidsiska-vega / @barnabee  I'm actually not sure how the orchestration here should work. Trading the full amount of the iceberg order probably makes sense technically but would conflict with the assertion that only the visible component can ever trade once the order is on the book. However if we only trade the visible component and there is still an order of the other side sitting on the book there will be further trades each time there is a refresh, and who would be the agressor in those trades?

Reply:
@TomMcL my take would be either:
a) make icebergs always post-only; in this case we would reject the order. 
b) make icebergs with post-only as on option; in this case I would trade potentially the full volume (both visible and hidden) if there is enough volume already on the book and the iceberg is aggressive.

Reply:
@davidsiska-vega I'm not sure that fully resolves the scenario above thought, or I may be missing something. What I'm picturing is:

1. Party 1 places an iceberg order for price `x`, size `200`, visible size `10` is placed on the book at an empty price level
2. A limit order from party 2, happy to be a taker, for price `x`, size `100` is placed on the book and trades `10` as a taker
3. Party 2's `90` remaining is now sitting on the book at the end of the tx
4. Do we immediately refresh and match again, with party 2 still counted as the taker? That seems hard to keep track of. Though the alternative is presumably that the full size of an iceberg order can immediately trade with incoming limit orders which feels incorrect